Code/GraphMol/RascalMCES/RascalMCES.cpp

```diff
@@ -1126,15 +1126,25 @@ std::vector<RascalResult> findMCES(RascalStartPoint &starter,
                      opts.maxFragSeparation, opts.exactConnectionsMatch,
                      opts.equivalentAtoms, opts.ignoreBondOrders));
   }
+  // If 2 cliques are the same size, this sort puts the one with the smaller
+  // number of fragments first, which may have fewer atoms.
+  std::sort(results.begin(), results.end(), details::resultCompare);
 
```

Comment:
When `singleLargestFrag` is set, you are sorting `results` twice.
Can't you put this sort in an `else` on the `if (opts.singleLargestFrag) {` and include the comparison of the number of fragments in your sort on line 1134?

Reply:
I've moved the first sort to an else.  In the sort on 1134 (now 1130) there is only 1 fragment at this stage - singleLargestFrag has already been applied by the time it gets there.  That was in fact the original bug - having applied singleLargestFrag, there were then cliques of different sizes left and we only want to keep the largest of them.